src/client/providers/execInTerminalProvider.ts

```diff
@@ -38,8 +40,13 @@ function execInTerminal(fileUri?: vscode.Uri) {
         filePath = `"${filePath}"`;
     }
     const terminal = vscode.window.createTerminal(`Python`);
+    if (pythonSettings.terminal.executeInFileDir) {
```

Comment:
I made this change (pushing shortly), but I'm unclear on how exactly that would be possible. The presence of a default value for this setting does not preclude it from not existing?